File: src/vta/inst_queues.cc
#include "vta/inst_queues.hh"

#include "base/trace.hh"
#include "debug/BaseVTAFlag.hh"

namespace gem5
{
LoadQueue::LoadQueue(const Params &params) : BaseInstructionQueue(params)
{
    DPRINTF(BaseVTAFlag, "create LoadQueue!\n");
}
} // namespace gem5

namespace gem5
{
ComputerQueue::ComputerQueue(const Params &params) :
    BaseInstructionQueue(params)
{
    DPRINTF(BaseVTAFlag, "create ComputerQueue!\n");
}
} // namespace gem5

namespace gem5
{
StoreQueue::StoreQueue(const Params &params) : BaseInstructionQueue(params)
{
    DPRINTF(BaseVTAFlag, "create StoreQueue!\n");
}
} // namespace gem5

File: src/vta/InstQueues.py
from m5.params import *
from m5.SimObject import SimObject


class BaseInstructionQueue(SimObject):
    type = "BaseInstructionQueue"
    abstract = True
    cxx_header = "vta/base_instruction_queue.hh"
    cxx_class = "gem5::BaseInstructionQueue"


class LoadQueue(BaseInstructionQueue):
    type = "LoadQueue"
    abstract = False
    cxx_header = "vta/inst_queues.hh"
    cxx_class = "gem5::LoadQueue"


class ComputerQueue(BaseInstructionQueue):
    type = "ComputerQueue"
    abstract = False
    cxx_header = "vta/inst_queues.hh"
    cxx_class = "gem5::ComputerQueue"


class StoreQueue(BaseInstructionQueue):
    type = "StoreQueue"
    abstract = False
    cxx_header = "vta/inst_queues.hh"
    cxx_class = "gem5::StoreQueue"
